src/main/java/Transcription/TranscriptionCache.java

```diff
@@ -0,0 +1,29 @@
+package Transcription;
+
+import Search.Searchable;
+
+public class TranscriptionCache {
+    private VoskAdapter speechToText;
+    private String filename;
+    private VoskProxy voskProxy;
+
+    public TranscriptionCache(String filename, VoskAdapter speechToText) {
+        this.filename = filename;
+        this.speechToText = speechToText;
+        this.voskProxy = new VoskProxy();
+    }
+
+    public Searchable getSearchable() {
+        CacheKey cacheKey = new CacheKey(filename, speechToText);
+        if (cacheKey.cachedFilesExist()) {
```

Comment:
The cacheKey should not have this method. This method is an implementation detail of TranscriptionCache, so put it there instead.

Reply:
Did this, also replaced with if(cachedFilesExist) rather than if(!cachedFilesExist). Is this a special case where the negative conditional makes more sense? If the cached files do not exist, query the speech to text?